apps/wallet/src/ui/app/hooks/useGetValidators.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { isSuiObject, isSuiMoveObject } from '@mysten/sui.js';
+import BigNumber from 'bignumber.js';
+import { useMemo } from 'react';
+
+import { useGetObject } from './useGetObject';
+
+//TODO: Remove when available on SDK, types should come from the SDK
+import type {
+    Validator,
+    ValidatorState,
+} from '_app/staking/home/ValidatorDataTypes';
+
+export const VALIDATORS_OBJECT_ID = '0x05';
+
+function processValidators(
+    set: Validator[],
+    totalStake: bigint,
+    current_epoch: number,
+    walletAddress?: string | null
+) {
+    return set.map((av) => {
+        const rawName = av.fields.metadata.fields.name;
+
+        const name = Buffer.from(rawName, 'base64').toString();
+
+        const {
+            sui_balance,
+            starting_epoch,
+            pending_delegations,
+            delegation_token_supply,
+        } = av.fields.delegation_staking_pool.fields;
+        const num_epochs_participated = current_epoch - starting_epoch;
+        const APY =
+            (1 +
+                (sui_balance - delegation_token_supply.fields.value) /
+                    delegation_token_supply.fields.value) ^
+            (365 / num_epochs_participated - 1);
+
+        const pending_delegationsByAddress = pending_delegations
+            ? pending_delegations.filter(
+                  (d) => d.fields.delegator === walletAddress
+              )
+            : [];
+
+        return {
+            name: name,
```

Comment:
I get __some__ of this extraction (e.g. `name` where it needs custom processing), but a lot of this seems like you could just return the entire `av` object, and do a lot of this processing where you need it in the UI. I'm very unconvinced that this hook really should exist. The pending delegation is a good example of something that I think is good to be a custom hook because it's a bit complex and also it's also not even generalized, it's just a helper hook within the staking code.

I'd probably recommend doing a lot more of this computation within the UI, and just making this return either the entire state object, or the `active_validators` with some light metadata.